PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/BaseSequence.java

```diff
@@ -0,0 +1,170 @@
+package nl.tudelft.pl2016gr2.model;
+
+import nl.tudelft.pl2016gr2.thirdparty.testing.utility.TestId;
+
+/**
+ * This is a container class for the bases of a genome. It efficiently encodes the ATCG and N
+ * (unknown) bases as bits in an array of integers, so fewer memory is needed than when using a
+ * simple String.
+ *
+ * @author Faris
+ */
+public class BaseSequence {
+
+  private static final int BITS_PER_BASE = 3;
+  private static final int BITS_PER_INT = Integer.BYTES * 8;
+
+  private static final int END_OF_BASES = 0b000;
+  private static final int BASE_A = 0b001;
+  private static final int BASE_T = 0b010;
+  private static final int BASE_C = 0b011;
+  private static final int BASE_G = 0b100;
+  private static final int BASE_UNKNOWN = 0b101;
+
+  @TestId(id = "bases")
+  private final int[] bases;
+
+  /**
+   * Create a base sequence.
+   *
+   * @param bases the sequence of bases.
+   */
+  public BaseSequence(String bases) {
+    this(bases.toCharArray(), 0, bases.length());
+  }
+
+  /**
+   * Create a base sequence.
```

Comment:
I had to look at the parser to understand the parameters of the constructor. Please clarify that you create a base sequence _based on the graph file char array and its indices_.